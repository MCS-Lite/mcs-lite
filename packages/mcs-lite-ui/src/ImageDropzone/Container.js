// @flow
import * as React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import P from '../P';

type Props = {
  src?: string,
  canDrop: boolean,
  height: number,
};

const Container: React.ComponentType<Props> = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: ${props => props.height}px;
  padding: 12px;
  text-align: center;
  box-sizing: border-box;

  background-color: ${props => props.theme.color.grayLight};
  border: 1px ${props => (props.src && !props.canDrop ? 'solid' : 'dashed')}
    ${props =>
      props.canDrop ? props.theme.color.primary : props.theme.color.grayBase};
  border-radius: 3px;
  transition: all cubic-bezier(0.47, 0, 0.75, 0.72) 0.3s;

  ${props =>
    props.src &&
    css`
      background-image: url("${props.src}");
      background-repeat: no-repeat;
      background-size: cover;
      background-position: center center;
    `};

  > ${P} {
    margin-top: 5px;
  }
`;

Container.displayName = 'Container';
Container.propTypes = {
  canDrop: PropTypes.bool.isRequired,
  src: PropTypes.string,
};

export default Container;
