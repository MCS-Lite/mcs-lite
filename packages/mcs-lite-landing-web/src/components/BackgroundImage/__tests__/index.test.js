import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Container from '../';

it('should render Container correctly with HOC', () => {
  const wrapper = shallow(<Container src="src" />);

  expect(toJson(wrapper)).toMatchSnapshot();
});
