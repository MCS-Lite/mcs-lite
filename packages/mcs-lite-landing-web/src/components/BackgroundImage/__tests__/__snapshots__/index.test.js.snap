// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Container correctly with HOC 1`] = `
<Styled(ProgressiveImage)
  blur={0.5}
  opacity={1}
  src="src"
  transition="blur 0.3 linear"
/>
`;
